enes-homework-week3-moreJS.js

```diff
@@ -0,0 +1,200 @@
+//Q:More JS
+
+//Q1
+
+function sum(a,b,c) {
+  return (a+b+c);
+};
+
+console.log(sum(1,2,3));
+
+//Q2
+
+function colorCar(str) {
+  console.log("a " + str + " car" );
+}
+
+const str1 = "red";
+
+console.log(colorCar(str1));
+
+
+//Q3
+const obj1 = {
+  name: "Jason",
+  surname: "Born",
+  age: 33,
+};
+
+function printOutPropertiesAndValues (obj) {
+  for (let key in obj) {
+    console.log(key);
+    console.log(obj[key]);
+  }
+}
+
+printOutPropertiesAndValues(obj1);// expected output: name surname age Jason Born 33
+
+Q4
+let vehicles = "";
+function codeDesigner(color, code) {//since one function is designed to do one job, I divided the question in two functions.
+  if (code === 1){
+    vehicles = "car";
+  } else if (code === 2){
+    vehicles = "motorcycle";
+  } else {
+    vehicles = "";
+    console.log("please select 1 for cars and 2 for motorcycles");
+  }
+}
+function vehicleType(color, code) {
+  codeDesigner(color, code);
+  console.log("a " + color + " " + vehicles);
+}
+
+vehicleType("blue", 1);
+
+  /* Another method: 
+  (However I could not find out how to write down  
+    "please select 1 for cars and 2 for motorcycles" 
+    when selected other than 1,2 with this method.)
+  function vehicleType(color, code) {
+    vehicles = (code ===1 ) ? "car" :
+              (code === 2 ) ? "motorcyle" : 
+              "";
+  console.log("a " + color + " " + vehicles);
+  }
+  */
+
+vehicleType("blue", 1);
+
+
+//Q5
+
+console.log((3===3) ? "yes" : "no"); //Since .log is an function it is possible to write down an expression in it.
+
+//Q6 
+/* function vehicle() necessitates the first two functions.
+let vehicles = "";
+function codeDesigner(color, code) {
+  if (code === 1){
+    vehicles = "car";
+  } else if (code === 2){
+    vehicles = "motorcycle";
+  } else {
+    vehicles = "";
+    console.log("please select 1 for cars and 2 for motorcycles");
+  }
+}
+function vehicleType(color, code) {
+  codeDesigner(color, code);
+  console.log("a " + color + " " + vehicles);
+}
+*/
+let situation = "";
+function vehicle(color, code, age){
+  codeDesigner(color, code);
+  situation = (age == 0) ? "brand-new" : "used";
```

Comment:
`==` or `===`? 😉

Reply:
And how about using the back-ticks in console log?

Reply:
surely ===. and console.log(`a ${color} ${situation} ${vehicles});